src/ImageSharp/Formats/Png/PngEncoderCore.cs

```diff
@@ -654,6 +655,38 @@ private void WriteExifChunk(Stream stream, ImageMetadata meta)
             this.WriteChunk(stream, PngChunkType.Exif, meta.ExifProfile.ToByteArray());
         }
 
+        /// <summary>
+        /// Writes an iTXT chunk, containing the XMP metdata to the stream, if such profile is present in the metadata.
+        /// </summary>
+        /// <param name="stream">The <see cref="Stream"/> containing image data.</param>
+        /// <param name="meta">The image metadata.</param>
+        private void WriteXmpChunk(Stream stream, ImageMetadata meta)
+        {
+            const int iTxtHeaderSize = 5;
+            if (((this.options.ChunkFilter ?? PngChunkFilter.None) & PngChunkFilter.ExcludeTextChunks) == PngChunkFilter.ExcludeTextChunks)
+            {
+                return;
+            }
+
+            if (meta.XmpProfile is null)
+            {
+                return;
+            }
+
+            var xmpData = meta.XmpProfile.Data;
+
+            if (xmpData.Length == 0)
+            {
+                return;
+            }
+
+            byte[] payload = new byte[xmpData.Length + PngConstants.XmpKeyword.Length + 5];
```

Comment:
This should be allocated using our memory allocator

Reply:
Indeed

Reply:
@JimBobSquarePants Test results on Github vary now, for the same Png round trip test case. I do NOT get this same error locally.

Can it be that this last memory optimization is causing instability? If so, I do not understand what might be the cause.

Reply:
Found it, the Memory Allocator does not zero out the buffers it allocates.
Doing this explitly now.

Btw: I found some unrelated methods in `PngEncoderCore` that suffer from the same allocation issue. I'll fix these also.